module_utils/network_lsr/argument_validator.py

```diff
@@ -2012,3 +2083,114 @@ def _ipv6_is_not_configured(connection):
                             "match.path is not supported by the running version of "
                             "NetworkManger.",
                         )
+        self.validate_route_tables(connection, idx)
+
+
+class IPRouteUtils(object):
+
+    ROUTE_TABLE_ALIAS_RE = re.compile("^[a-zA-Z0-9_.-]+$")
+
+    @classmethod
+    def _parse_route_tables_mapping(cls, file_content, mapping):
+        for line in file_content.split(b"\n"):
+
+            #
+            # We are a bit more forgiving, but the aliases must be nice UTF-8 names.
+
+            # iproute2 only skips over leading ' ' and '\t'. We are more forgiving,
+            # and strip all leading+trailing whitespace.
```

Comment:
What is the reason to deviate from iproute2, here?

Reply:
Please note that `str.lstrip()` exists to skip over leading whitespace and it also accepts the characters to be stripped as a parameter.

Reply:
Originally I was tending to be more fault tolerant for the user when they added their custom route tables either through the Ansible lineinfile module or other methods. But after the second thought, I realized that this may inadvertently allow the user adding the table which can be parsed by the role but can not be parsed by iproute2.